exercises/08-exercise.py

```diff
@@ -31,26 +33,62 @@ def __init__(self, resource_path, chunk_size=1024):
             user specifices size of chunk for data, default is 1024
         '''
         self.chunks_of_data = []
-        for chunk in pd.read_csv(resource_path, chunksize=chunk_size, 
-                names =['time (s)', 'time (ns)', 'index', 'counts'], 
-                delimiter = " ", header=None):
-            chunk['adc'] = chunk['counts'].apply(adc2counts)
+        chunk = [data for data in pd.read_csv(resource_path, 
+            chunksize=chunk_size, delimiter = " ", header = None) ]
+        
+        num_cols = len(chunk[0].columns)
+
+        columns = ['time (s)', 'time (ns)', 'index']
+        columns_leftover = num_cols - len(columns)
+        columns = columns + [f'adc {i}' for i in range(columns_leftover)]
+        
+        for chunk in chunk:
+            chunk.columns = columns
+            
+            for column in range(columns_leftover):
+                chunk.iloc[:, column + 3] = chunk.iloc[:, column + 3].apply(adc2counts)
+            
             chunk['timestamp'] = chunk['time (s)'] + 1e-9*chunk['time (ns)']
-            chunk = chunk.drop(columns = ['time (s)', 'time (ns)', 'index', 'counts'])
-            chunk = chunk[['timestamp','adc']]
-            self.chunks_of_data.append(chunk)
-
-    def __call__(self, chunk_num):
+            chunk = chunk.drop(columns = ['time (s)', 'time (ns)', 'index'])
+            new_cols = chunk.columns.tolist()
+            new_cols = new_cols[-1:] + new_cols[:-1]
```

Comment:
I don't know if it's a good idea to guarantee that 'timestamp' will always be the last column.
How about index them by key since we know them? Then we're assured there's no ambiguity
```py3
column_keys = ['timestamp'] + [f'adc {i}' for i in range(self.columns_leftover)]
chunk = chunk[column_keys]
```
(we can then ignore the `drop` step)

Reply:
Thanks! Much cleaner way!